### PR TITLE
implement support for erc20

### PR DESCRIPTION
**Pull request by vadshalamov**
https://github.com/rubykube/peatio/issues/384

### CODE REVIEW

**Comment by mod** [comment]
That's a good start ^^
You missing unit tests though

File: lib/coin_api/eth.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+module CoinAPI
+  class Eth < BaseAPI
```
Comment (yivo) [inline lib/coin_api/eth.rb]: @vadshalamov The convention is to use `ETH`.

**Review by yivo — CHANGES_REQUESTED** [review]

**Comment by vadshalamov** [comment]
@mod @yivo  how should I handle if user want to connect several token addresses(contract addresses)? or user will have the opportunity to work with only one contract address and wallet address at one time?

File: lib/coin_api/json_rpc/base_api.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module CoinAPI
+  module JsonRPC
+    class BaseAPI < ::CoinAPI::BaseAPI
```
Comment (yivo) [inline lib/coin_api/json_rpc/base_api.rb]: @vadshalamov Consider moving this stuff to ETH.

File: .rubocop.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+AllCops:
```
Comment (yivo) [inline .rubocop.yml]: @vadshalamov Rubocop is not in scope of this PR.

File: Gemfile [hunk 4]
```diff
@@ -84,6 +84,7 @@ end
 group :test do
   gem 'database_cleaner'
   gem 'mocha', require: false
+  gem 'webmock'
```
Comment (yivo) [inline Gemfile]: @vadshalamov Consider locking gem version range.

**Review by yivo — CHANGES_REQUESTED** [review]

**Comment by yivo** [comment]
@vadshalamov We are not very familiar with all ETH stuff. Can you push the simplest implementation you think?

**Comment by vadshalamov** [comment]
@yivo up

**Pull request closed without merging**